### ISSUE TITLE
Can AccelSim simulate QV100 using Trace generated by non-Volta architecture GPUs?

### ISSUE THREAD

**Issue by Rhythmicc** [accel-sim/accel-sim-framework#281]
I'm trying to generate a trace of the sample program using non-Volta GPUs such as TITANX, TITAN RTX and A100, however I'm getting the assertion error with the following command:
```
> ./util/job_launching/run_simulations.py -C QV100-SASS -B custom_apps -T ./hw_run/traces/device-1/11.0 -N t1

...
accel-sim.out: shader.cc:1473: void scheduler_unit::cycle(): Assertion `spec_id < m_shader->m_config->m_specialized_unit.size()' failed.
```

I noticed in #144 that it states that a Volta architecture GPU should be used to generate the traces before simulation, but I don't have a Volta architecture GPU on hand, and I was wondering if it's possible to configure something in a way that would allow traces generated by GPUs of other architectures to be simulated by the QV100 configuration?

**Comment by JRPan** [comment]
You can use any GPU you want. The requirement is you have to **use the same config with the GPU that you collected trace with**. So TITAN generated traces -> uses TITAN config. TITAN RTX -> Turing config. A100 -> Ampere config.

We have configs for RTX2060 and RTX3070. You can created one for ur GPU with the tuner.

**Comment by Rhythmicc** [comment]
Thank you very much! Also I would like to ask: if I want to replace the Tensor Core with new hardware in the GPU and add new instructions, is there any documentation to support me in doing this?🙏

**Comment by LohitDaksha** [comment]
Hi @Rhythmicc, were you able to model your custom architecture??

**Comment by Rhythmicc** [comment]
> [@Rhythmicc](https://github.com/Rhythmicc) Hey! were you able to model your custom architecture??

No, I took a compromise approach: I implemented a simulator for Tensor Core (TC) using C++ and replaced all TC call instructions in the trace with NOPs. This allowed AccelSim to complete all simulations outside TC, and by combining it with the results from our simulator, we obtained the final performance. We compared it with the hardware at hand, and the error is acceptable.

**Comment by LohitDaksha** [comment]
Thanks! @Rhythmicc
